naming-conventions.md

```diff
@@ -226,6 +226,53 @@ This will produce the following JSON:
 }
 ```
 
+### Request/Response types
+
+For complicated `Handler` requests or responses, where we use a custom type
+instead of an `Entity` directly, such types should be named `{Resource}{Method}`
```

Comment:
I also think another acceptable exception is data types that have some obvious name unrelated to being a `GET` response, e.g `ElaReadingReport`. Especially if you're doing a bunch of pure logic to generate one, the fact that it's being used as a `GET` response doesn't feel important enough to reflect in the naming to me.

Reply:
That's a good point. I agree we should soften the language to not let this guide preclude making a proper domain object that can also serve as the request/response type.

Maybe instead of saying:

> a custom type instead of an `Entity` directly, ...

We can say

> a custom type instead of an existing domain object, ...

In the `Report` case, I would call that a generally-useful domain object that just happens to only be used to support a response at the moment.

Reply:
Yeah, I like the domain object language